### PR TITLE
fix(datetime): improve value normalization and prevent ionic parsing warnings

### PR DESCRIPTION
**Pull request by felipefialho**
This pull request enhances the `AtomDatetime` component by improving its handling of various input values, refining its internal logic, and updating tests to ensure robustness. The key changes include better handling of empty or invalid input values, improvements to date sorting and formatting, and updates to ensure proper behavior of min/max and multiple/range mode properties.

### Enhancements to input handling:
* Introduced helper methods (`filterEmptyStrings`, `handleStringValue`, and `handleCustomEventValue`) to cleanly process empty strings, arrays, and custom event values. This ensures that invalid or empty inputs are gracefully handled. (`packages/core/src/components/datetime/datetime.tsx`, [packages/core/src/components/datetime/datetime.tsxR99-R139](diffhunk://#diff-283df32286ec2fd9e77ab2f8dd00915e18be2089f494c7c80b217aacf396de88R99-R139))
* Updated the `normalizeValue` method to leverage the new helper methods, simplifying logic and improving maintainability. (`packages/core/src/components/datetime/datetime.tsx`, [packages/core/src/components/datetime/datetime.tsxR99-R139](diffhunk://#diff-283df32286ec2fd9e77ab2f8dd00915e18be2089f494c7c80b217aacf396de88R99-R139))

### Improvements to date processing:
* Enhanced date sorting in `getRangeLabel` to correctly handle date strings by converting them to `Date` objects for accurate comparisons. (`packages/core/src/components/datetime/datetime.tsx`, [packages/core/src/components/datetime/datetime.tsxL306-R322](diffhunk://#diff-283df32286ec2fd9e77ab2f8dd00915e18be2089f494c7c80b217aacf396de88L306-R322))

### Refinements to property handling:
* Ensured `min` and `max` attributes are not passed to the DOM when their values are empty, by trimming and validating them. (`packages/core/src/components/datetime/datetime.tsx`, [packages/core/src/components/datetime/datetime.tsxL367-R384](diffhunk://#diff-283df32286ec2fd9e77ab2f8dd00915e18be2089f494c7c80b217aacf396de88L367-R384))
* Adjusted the `value` property to handle single and multiple date values correctly based on the `multiple` or `rangeMode` flags. (`packages/core/src/components/datetime/datetime.tsx`, [packages/core/src/components/datetime/datetime.tsxL383-R403](diffhunk://#diff-283df32286ec2fd9e77ab2f8dd00915e18be2089f494c7c80b217aacf396de88L383-R403))

### Testing updates:
* Added multiple test cases to validate the handling of empty strings, arrays, null/undefined values, and custom events. These tests ensure the component behaves as expected under various edge cases. (`packages/core/src/components/datetime/datetime.spec.ts`, [packages/core/src/components/datetime/datetime.spec.tsR320-R475](diffhunk://#diff-427b98ee76cf0d1838309648a7c4d252518c305a7cd53dba02a9e40f7229cebbR320-R475))

### Miscellaneous:
* Added a missing `React` import in the `datetime.react.stories.tsx` file to fix a potential issue. (`packages/core/src/components/datetime/stories/datetime.react.stories.tsx`, [packages/core/src/components/datetime/stories/datetime.react.stories.tsxR3](diffhunk://#diff-7ab3d609984aa5de31e5d81477c0f7f428096b07d795148577705e717ed0076cR3))

### CODE REVIEW

**Review by gabrielduete — DISMISSED** [review]

**Review by tassioFront — APPROVED** [review]

**Review by igorwessel — APPROVED** [review]

**Review by mauriciomutte — APPROVED** [review]

File: packages/core/src/components/datetime/stories/datetime.react.stories.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
+
```
[the remark this answers is not in the document]
Comment (felipefialho) [reply packages/core/src/components/datetime/stories/datetime.react.stories.tsx]: done at [741ab73](https://github.com/juntossomosmais/atomium/pull/777/commits/741ab73842dd73f3b274c45d539d9fdeda695ae3)

File: packages/core/src/components/datetime/datetime.tsx [hunk 2]
```diff
@@ -205,18 +228,18 @@ export class AtomDatetime {
     }, 0)
   }
 
-  private handleDateChange = (
+  readonly handleDateChange = (
```
[the remark this answers is not in the document]
Comment (felipefialho) [reply packages/core/src/components/datetime/datetime.tsx]: done at [4b6f985](https://github.com/juntossomosmais/atomium/pull/777/commits/4b6f985702966c52b578cada4ff6ed8b6a8cffb4)

File: packages/core/src/components/datetime/datetime.tsx [hunk 3]
```diff
@@ -96,31 +96,54 @@ export class AtomDatetime {
...
+
+    if (typeof detailValue === 'string') {
+      return this.handleStringValue(detailValue)
+    }
+
+    if (Array.isArray(detailValue)) {
+      return this.filterEmptyStrings(detailValue)
+    }
+
+    return undefined
```
[the remark this answers is not in the document]
Comment (felipefialho) [reply packages/core/src/components/datetime/datetime.tsx]: done at [3f8bc8a](https://github.com/juntossomosmais/atomium/pull/777/commits/3f8bc8a495267f3306f890d9ca9e3aafb2db80a5)

File: packages/core/src/components/datetime/datetime.tsx [hunk 4]
```diff
@@ -364,8 +389,8 @@ export class AtomDatetime {
         id={this.datetimeId}
         isDateEnabled={this.isDateEnabled}
         locale={this.locale}
-        max={this.max}
-        min={this.min}
+        max={this.max && this.max.trim() !== '' ? this.max : undefined}
+        min={this.min && this.min.trim() !== '' ? this.min : undefined}
```
[the remark this answers is not in the document]
Comment (felipefialho) [reply packages/core/src/components/datetime/datetime.tsx]: done [84fd296](https://github.com/juntossomosmais/atomium/pull/777/commits/84fd296be2c4c698aecc65dd6f7e5478a3705f7b)
